src/main/java/net/minecraftforge/client/event/DimensionLightMapModificationEvent.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import net.minecraft.util.math.vector.Vector3f;
+import net.minecraft.world.World;
+
+/**
+ * Allows modders to adjust the lightmap colours for a specific Dimension
+ * <br>
+ * While DimensionRenderInfo could look like the closest solution for this task, its entries are stored in private, immutable map that is specific to a DimensionType
+ * <br>
+ * There can be many Dimension(s) per DimensionType, which means DimensionRenderInfo is limited in both functionality and encourages poor practices.
+ * <br>
+ * Hence, this event is the preferred solution over DimensionRenderInfo.
+ * <br>
+ * This event fired in {@link LightMapTexture#updateLightMap}
+ */
```

Comment:
A suggestion for the javadoc, just to tidy it up a bit more (and use less `<br>` tags):
```java
/**
 * <p>Allows modders to adjust the lightmap colours for a specific {@code Dimension}.</p>
 * 
 * <p>While {@code DimensionRenderInfo} could look like the closest solution for this task, 
 * its entries are stored in private, immutable map that is specific to a {@code DimensionType}. 
 * There can be many {@code Dimension}(s) per {@code DimensionType}, which means 
 * {@code DimensionRenderInfo} is limited in both functionality and encourages poor practices.</p>
 * 
 * <p>Hence, this event is the preferred solution over DimensionRenderInfo.</p>
 * 
 * <p>This event is not {@linkplain net.minecraftforge.eventbus.api.Cancelable cancellable}, 
 * and does not {@linkplain HasResult have a result}.</p>
 * 
 * <p>This event fired in {@link LightMapTexture#updateLightMap}, on the 
 * {@link net.minecraftforge.common.MinecraftForge#EVENT_BUS} only on the 
 * {@linkplain net.minecraftforge.fml.LogicalSide#CLIENT client-side}.
 */
```

Reply:
Thanks! I like your suggestion, it covers a few things I did not consider. I will include it in my next commit.